src/UI/Art/SpriteArtist/MiniMapArtist.java

```diff
@@ -101,6 +109,66 @@ private static BufferedImage generateMapImage(MapInfo mapInfo, Color[] teamColor
     return image;
   }
 
+  /**
+   * Generates a BufferedImage containing the full-scale map.
+   */
+  private static BufferedImage generateFullMapImage(MapInfo mapInfo, Color[] teamColors)
```

Comment:
I assume you considered it, but were you not able to just have `SpriteMiniMapArtist` draw itself instead of reimplementing this logic? That seems like the obvious thing to do, though admittedly I haven't looked to see how feasible that would be.

Reply:
1) It's all tied up in the `GameMap` interface proper.
2) It's spread across no less than 2 functions and 2 classes, and would probably require faking a set of COs and `GameRules`.

I'll warrant it's a little ugly to have it duplicated, but I really don't see a way to fix that.

I'll also warrant that mixing the true-scale and 1:1-scale drawing logic isn't great either, but you can't really get what I was looking for without making that jump.